### PR TITLE
Registry down crippling fix + shrink feature

### PR DESCRIPTION
**Pull request by NorseGaud**
- Pull can fail and not fail the entire pipeline. This helps prevent the registry going down from crippling the pipeline. Updated readme.
- Added "shrink" for always-pull so you can remove tags that aren't needed anymore.
- Various cleanup and simplifications.
- Added tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +89,7 @@ Example: `true`
 ### `cleanup` (optional)
 
 Set this to `false` to leave the cloned images in a failed build for investigation.
+- You will need to run your buildkite agent with `cancel-grace-period=60`, as the [default 10 seconds is not enough time](https://forum.buildkite.community/t/problems-with-anka-plugin-and-pre-exit/365/7).
```
Comment (tduffield) [inline README.md]: So, something we could do is provide some log output when we detect this setting is too low. From out plugin we _should_ be able to inspect `$BUILDKITE_CANCEL_GRACE_PERIOD` and provide some log output that says something to the effect of "if you see weird suspension behavior, please set this to FOO"

Comment (NorseGaud) [reply README.md]: Good idea. However, I'm not seeing $BUILDKITE_CANCEL_GRACE_PERIOD set. I tried finding it within the pipeline.yml command and also within the plugin with no luck. Any ideas on why that is?

Comment (tduffield) [reply README.md]: Hmm, I guess its a way to set the value, but not read it. That's too bad. We should think about this functionality in theory, but let's not block this work on it.

Comment (NorseGaud) [reply README.md]: Roger that. I'm all done with the commits for now.

**Pull request merged**